### PR TITLE
l2geth: remove ovmsigner

### PR DESCRIPTION
**Pull request by tynes**
**Description**
Removes the `OVMSigner`, required for RLP transaction regenesis

### CODE REVIEW

File: l2geth/core/types/transaction_signing.go [hunk 1]
```diff
@@ -91,7 +40,16 @@ type sigCache struct {
 
 // MakeSigner returns a Signer based on the given chain config and block number.
 func MakeSigner(config *params.ChainConfig, blockNumber *big.Int) Signer {
-	return NewOVMSigner(config.ChainID)
+	var signer Signer
+	switch {
+	case config.IsEIP155(blockNumber):
+		signer = NewEIP155Signer(config.ChainID)
+	case config.IsHomestead(blockNumber):
+		signer = HomesteadSigner{}
```
Comment (smartcontracts) [inline l2geth/core/types/transaction_signing.go:48]: Do we want to support `HomesteadSigner` and `FrontierSigner`? If so, why?

**Review by smartcontracts — CHANGES_REQUESTED** [review]
Looks good, just left one question. Will approve once that's answered.

File: l2geth/core/types/transaction_signing.go (hunk 1 above)
Comment (karlfloersch) [reply l2geth/core/types/transaction_signing.go:48]: Is the reason for supporting those is just minimal diff? I'd presume that's why

**Review by karlfloersch — COMMENTED** [review]
+1 to @smartcontracts 's question, but if my intuition for why you did it is correct (minimal diff) I think this is good to go! Being able to revert code back to the original format is great

File: l2geth/core/types/transaction_signing.go (hunk 1 above)
Comment (tynes) [reply l2geth/core/types/transaction_signing.go:48]: The only reason why they are there is for minimal diff. I'd prefer to just:
```golang
reutnr newEIP155Signer(config.ChainID)
```

Comment (smartcontracts) [reply l2geth/core/types/transaction_signing.go:48]: Ok yea I think it's better to make the diff here

Comment (gakonst) [reply l2geth/core/types/transaction_signing.go:48]: IMO we should always be striving for a minimal diff, even in places where it seems trivial. There's a lot of dead code which we're not using (e.g. p2p stuff, ethash etc.), but we haven't removed it, so we should not remove this part either, and try to stay as close to geth as possible.

**Review by gakonst — APPROVED** [review]

**Review by smartcontracts — APPROVED** [review]

**Comment by tynes** [comment]
While using clique consensus, EIP155 is enabled from the start so we should always get the `EIP155Signer`

**Pull request merged**